ts/Extensions/Annotations/Annotations.ts

```diff
@@ -1359,50 +1359,11 @@ merge<Annotation>(
                  *         Attach annotation to a mock point
                  *
                  * @declare   Highcharts.AnnotationMockPointOptionsObject
-                 * @type      {string|*}
+                 * @type      {string|Highcharts.AnnotationMockPointOptionsObject}
```

Comment:
This can also be `AnnotationMockPointFunction` (like in the new demo)

Reply:
Then it can not declare Highcharts.AnnotationMockPointOptionsObject automatically. You have to define that interface yourself.